components/server/ee/src/prebuilds/bitbucket-app.ts

```diff
@@ -112,11 +121,16 @@ export class BitbucketApp {
             }
 
             const contextURL = this.createContextUrl(data);
-            const context = (await this.contextParser.handle({ span }, user, contextURL)) as CommitContext;
             span.setTag("contextURL", contextURL);
+            const context = (await this.contextParser.handle({ span }, user, contextURL)) as CommitContext;
+            await this.webhookEvents.updateEvent(event.id, {
+                projectId: projectAndOwner?.project?.id,
+                cloneUrl: context.repository.cloneUrl,
+            });
             const config = await this.prebuildManager.fetchConfig({ span }, user, context);
             if (!this.prebuildManager.shouldPrebuild(config)) {
                 console.log("Bitbucket push event: No config. No prebuild.");
+                await this.webhookEvents.updateEvent(event.id, { status: "ignored_unconfigured" });
```

Comment:
Hm. As the table is about WebhookEvents now this prebuild-specific status does not make sense now: Imagine a webhook triggered both, a preview and a prebuild, what status would it be in?

What about keeping the `status` strictly with the WebHook:
```
"received"
  | "dismissed_unauthorized"
  | "ignored_incomplete"
  | "processed" // or similar
```

The link to the prebuild exists via `prebuildId`. Still, we need a place to document the decisions, e.g. whether or not a prebuild got triggered... :thinking:

Reply:
> prebuild-specific status does not make sense now: Imagine a webhook triggered both, a preview and a prebuild, what status would it be in?

Good question. I think instead of generalizing to a minimal set of commons, I'd rather split the status by concerns. In fact, we need explicit states for the frontend to render. If we coerce different concerns into generic states, I fear it will be difficult to render meaningful states. 

What about adding explicit `prebuildStatus`? We can tag specific statuses. 

Also, is there something to read more on the new preview feature? I find it hard to reason about, as I've no details so far.

Reply:
> I believe the webhook table should not contain any prebuild information at all.

It's certainly a trade off, but I feel that it's worth it. The goal here is to untangle the question "what happened to this prebuild event and why" from the `PrebuiltWorkspace` state machine (which feels broken atm).

We could certainly add another table, e.g. WebhookEvent -> PrebuildEvent -> PrebuiltWorkspace, but as we'd join that on primary key of the WebHook event and to avoid the additional complexity here, I think it's ok (for now) to have that inside the WebhookEvent table. :lotus_position: